obal/data/roles/fetch_sources/tasks/main.yml

```diff
@@ -74,60 +74,53 @@
         spec_file_path: "{{ spec_file.files[0].path }}"
         setup_sources_flags: '-f'
 
-    - name: 'Set downstream release'
+    - name: 'Set release'
       set_fact:
-        downstream_release: '1'
+        release: '1'
       when:
-        - downstream_release is not defined
+        - release is not defined
 
-    - name: 'Set default downstream changelog'
+    - name: 'Set default changelog'
       set_fact:
-        downstream_changelog: '- Release {{ inventory_hostname }} {{ downstream_version }}'
+        changelog: '- Release {{ inventory_hostname }} {{ version }}'
       when:
-        - downstream_changelog is not defined
+        - changelog is not defined
+        - version is defined
 
     - name: 'Ensure Source0 specfile points to source server'
       replace:
         path: "{{ spec_file_path }}"
         regexp: '^(?P<prefix>Source0:\s+)http.*/(?P<filename>.*)$'
         replace: '\g<prefix>{{ source_server }}\g<filename>'
+      when: source_server is defined
```

Comment:
`source_server` tends to be defined on a group/global level, so this changes the `Source0` line even on "fetch from upstream" scenarios. Should this test for `source_server` *and* `version` being defined?

Reply:
The implication being if version and source_server are defined then assume that strategy? Would you ever do one without the other? Sounds like we may need to re-think this in general around defined strategies. For now, it cna help avoid issues so I'll update.

Reply:
Yeah. At some point, we might think about adding a `strategy` switch, defaulting to `upstream`, and behave differently when set to `downstream` (this applies at least to `add` and `update` actions).

But for now the implicit solution is good enough for me :)